imap-proto/src/types.rs

```diff
@@ -81,6 +81,14 @@ pub enum MailboxDatum<'a> {
     Recent(u32),
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum Capability<'a> {
+    IMAP4rev1,
+    /// Defined by [RFC2222 - SASL](https://tools.ietf.org/html/rfc2222)
```

Comment:
Not sure what this comment means. Seems to me that `AUTH` is defined in RFC 3501, so how is RFC 2222 relevant here? And if it is relevant to one of the variants, the comment should probably be on the same line as that comment?